Samples/MobileBuyIntegration/MobileBuyIntegration/ProductView.swift

```diff
@@ -0,0 +1,290 @@
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+*/
+
+import Buy
+import UIKit
+import SwiftUI
+import ShopifyCheckoutSheetKit
+
+struct ProductView: View {
+    // MARK: Properties
+    @State private var product: Storefront.Product
+    @State private var handle: String?
+    @State private var loading = false
+	@State private var imageLoaded: Bool = false
+    @State private var showingCart = false
+    @State private var descriptionExpanded: Bool = false
+    @State private var addedToCart: Bool = false
+
+    init(product: Storefront.Product) {
+        _product = State(initialValue: product)
+    }
+
+    // MARK: Body
+    var body: some View {
+        ScrollView {
+            VStack(spacing: 16) {
+                 if let imageURL = product.featuredImage?.url {
+					ZStack {
+						Rectangle()
+							.fill(Color.gray.opacity(0.2))
+							.frame(height: 400)
+
+						AsyncImage(url: imageURL) { phase in
+							switch phase {
+							case .empty:
+								EmptyView()
+							case .success(let image):
+								image
+									.resizable()
+									.aspectRatio(contentMode: .fill)
+									.frame(height: 400)
+									.clipped()
+									.opacity(imageLoaded ? 1 : 0)
+									.onAppear {
+										withAnimation(.easeIn(duration: 0.5)) {
+											imageLoaded = true
+										}
+									}
+							case .failure:
+								Image(systemName: "photo")
+									.resizable()
+									.frame(width: 100, height: 100)
+									.foregroundColor(.gray)
+							@unknown default:
+								EmptyView()
+							}
+						}
+					}
+					.frame(height: 400)
+				}
+
+                VStack(alignment: .leading, spacing: 8) {
+
+                    Text(product.vendor)
+                        .font(.body)
+                        .fontWeight(.semibold)
+						.padding(.vertical)
+                        .foregroundColor(Color(ColorPalette.primaryColor))
+
+					Text(product.title)
+                        .font(.title)
+
+                    Text(product.description)
+                        .font(.body)
+                        .foregroundColor(.gray)
+                        .lineLimit(descriptionExpanded ? 10 : 3)
```

Comment:
I think 10 for the expanded text is still bit too low for some of the products in the store (e.g. looking at Bird of paradise)

Reply:
Yeah, probably. I'll account for this in the full redesign task